### PR TITLE
Added JSON Schema for HLR V2

### PR DESCRIPTION
**Pull request by michelpmcdonald**
# New schema
Add new HLR V2 Schema
department-of-veterans-affairs/va.gov-team#28112

## Pull Requests to update the schema in related repositories

department-of-veterans-affairs/vets-website#18249
department-of-veterans-affairs/vets-api#7632

### CODE REVIEW

File: src/schemas/HLR-create-request-body_v1/schema.json [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                "informalConference",
+                "benefitType",
+                "veteran",
+                "socOptIn"
+              ],
+              "if": { "properties": { "informalConference": { "const":  true } } },
+              "then": { "required":  ["informalConferenceContact", "informalConferenceTime"] }
+            }
+          },
+          "additionalProperties": false,
```
Comment (michelpmcdonald) [inline src/schemas/HLR-create-request-body_v1/schema.json:125]: Draft-07 added explicit conditional schema evaluationI(if-then).  Current version of Ajv JSON schema validator used by this repo for unit test does not support Draft-07

File: src/schemas/HLR-create-request-body_v1/schema.json [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                },
+                "socOptIn":                  { "type": "boolean" }
+              },
+              "required": [
+                "informalConference",
+                "benefitType",
+                "veteran",
+                "socOptIn"
+              ],
+              "if": { "properties": { "informalConference": { "const":  true } } },
```
Comment (michelpmcdonald) [inline src/schemas/HLR-create-request-body_v1/schema.json:121]: Draft-07 added explicit conditional schema evaluationI(if-then). Current version of Ajv JSON schema validator used by this repo for unit test does not support Draft-07

File: src/schemas/HLR-create-request-body_v1/schema.json [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+              "then": { "required":  ["informalConferenceContact", "informalConferenceTime"] }
```
Comment (michelpmcdonald) [inline src/schemas/HLR-create-request-body_v1/schema.json:122]: Draft-07 added explicit conditional schema evaluationI(if-then). Current version of Ajv JSON schema validator used by this repo for unit test does not support Draft-07

File: test/common/all-schemas.spec.js [hunk 4]
```diff
@@ -27,7 +27,7 @@ describe('all schema tests', () => {
     if (!schemas.hasOwnProperty(k)) continue;
 
     // skip "checkObjectTypes" for these dist files used as enums
-    if (['definitions', 'constants', 'vaMedicalFacilities', 'caregiverProgramFacilities', 'form1010cgCertifications'].includes(k)) continue;
+    if (['HLR-CREATE-REQUEST-BODY_V1', 'definitions', 'constants', 'vaMedicalFacilities', 'caregiverProgramFacilities', 'form1010cgCertifications'].includes(k)) continue;
```
Comment (michelpmcdonald) [inline test/common/all-schemas.spec.js:30]: Draft-07 added explicit conditional schema evaluationI(if-then). Current version of Ajv JSON schema validator used by this repo for unit test does not support Draft-07 so I excluded the new schema from unit tests

**Review by ndsprinkle — APPROVED** [review]

**Pull request merged**